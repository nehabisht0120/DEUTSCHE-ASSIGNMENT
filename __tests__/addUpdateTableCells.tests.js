// this is test fole for tableCells.utils.js;

"use strict";

const tableUtils = require("../es6/tableUtils/tableCells.utils");
const rowUtils = require("../es6/tableUtils/tableRow.utils");

document.body.innerHTML =
  '<table id="stomp-table">' +
  "<tr>" +
  "<th>name</th>" +
  "<th>bestBid</th>" +
  "<th>bestAsk</th>" +
  "<th>openBid</th>" +
  "<th>openAsk</th>" +
  "<th>lastChangeAsk</th>" +
  "<th>lastChangeBid</th>" +
  "<th>SparkLines</th>" +
  "</tr>" +
  " </table>";

const mockedData = {
  name: "gbpeur",
  bestAsk: 1.3322889082436173,
  bestBid: 1.2810153021023876,
  lastChangeAsk: 0.020660545136841346,
  lastChangeBid: 0.009876039365910305,
  openAsk: 1.3030843400530694,
  openBid: 1.2729156599469307
};

const dataArr1 = [mockedData, mockedData];
const dataArr2 = [
  {
    name: "gbpeur",
    bestAsk: 1.3322889082436173,
    bestBid: 1.2810153021023876,
    lastChangeAsk: 0.020660545136841346,
    lastChangeBid: 0.009876039365910305,
    openAsk: 1.3030843400530694,
    openBid: 1.2729156599469307
  },
  {
    name: "euswq",
    bestAsk: 1,
    bestBid: 1,
    lastChangeAsk: 0.020660545136841346,
    lastChangeBid: 0.009876039365910305,
    openAsk: 1.3030843400530694,
    openBid: 1
  },
  {
    name: "euswq",
    bestAsk: 2,
    bestBid: 3,
    lastChangeAsk: 0.020660545136841346,
    lastChangeBid: 0.009876039365910305,
    openAsk: 1.3030843400530694,
    openBid: 4
  }
];

test("it should fill  coloumns value with for every unique stomp update", () => {
  // Set up our document body
  var row = rowUtils.InsertRow(mockedData, "stomp-table");
  tableUtils.addUpdateTableCells(mockedData, row);
  expect(row.innerHTML).toMatchSnapshot();
});

test("it should fill coloumns in correct row of the table", () => {
  var row = rowUtils.InsertRow(mockedData, "stomp-table");
  tableUtils.addUpdateTableCells(mockedData, row);
  expect(row.innerHTML).toMatchSnapshot();
  expect(row.id).toEqual(mockedData.name);
});

test("it should fill the coloumns in row with the latest updates", () => {
  
  //snapshot testing should fail in case you add another obj in dataArr2
  
  dataArr2.map(data => {
    var row = rowUtils.InsertRow(data, "stomp-table");
    tableUtils.addUpdateTableCells(data, row);
  });
  expect(document.getElementById("stomp-table")).toMatchSnapshot();
});
