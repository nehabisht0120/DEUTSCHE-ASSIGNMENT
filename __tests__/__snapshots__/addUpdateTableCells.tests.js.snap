// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should fill  coloumns value with for every unique stomp update 1`] = `"<td>gbpeur</td><td>1.28</td><td>1.33</td><td>1.27</td><td>1.30</td><td>0.02</td><td>0.01</td>"`;

exports[`it should fill coloumns in correct row of the table 1`] = `"<td>gbpeur</td><td>1.28</td><td>1.33</td><td>1.27</td><td>1.30</td><td>0.02</td><td>0.01</td>"`;

exports[`it should fill the coloumns in row with the latest updates 1`] = `
<table
  id="stomp-table"
>
  <tbody>
    <tr>
      <th>
        name
      </th>
      <th>
        bestBid
      </th>
      <th>
        bestAsk
      </th>
      <th>
        openBid
      </th>
      <th>
        openAsk
      </th>
      <th>
        lastChangeAsk
      </th>
      <th>
        lastChangeBid
      </th>
      <th>
        SparkLines
      </th>
    </tr>
     
    <tr
      id="gbpeur"
    >
      <td>
        gbpeur
      </td>
      <td>
        1.28
      </td>
      <td>
        1.33
      </td>
      <td>
        1.27
      </td>
      <td>
        1.30
      </td>
      <td>
        0.02
      </td>
      <td>
        0.01
      </td>
    </tr>
    <tr
      id="euswq"
    >
      <td>
        euswq
      </td>
      <td>
        3.00
      </td>
      <td>
        2.00
      </td>
      <td>
        4.00
      </td>
      <td>
        1.30
      </td>
      <td>
        0.02
      </td>
      <td>
        0.01
      </td>
    </tr>
  </tbody>
</table>
`;
